README.md

```diff
@@ -93,78 +254,120 @@ These segments are dropped:
+[Read more about this approach](http://cloudmark.github.io/Language-Detection/).
 
-For this project, the text will be split into lowercase characters, which are then combined to groups of size n, where n is a numeric value.
+For this project, the text will be split into lowercase characters, which are then re-combined to groups of size n, where n is a numeric value.
 These groups are then used further. 
 
-#### HashingTF
+##### HashingTF
 
-Unfortunately, Spark is not capable of machine learning using texts. These texts need to be transformed into numbers, where the easiest way is the use of a hashing function.
+Unfortunately, Spark is not capable of machine learning using texts.
+These texts need to be transformed into numbers, where the easiest way is the use of a hashing function.
 
-### Label preparation
+##### Label preparation
 
-[Labels][definition_labels] are the solution of a given problem.
-But since machine learning can only be applied on numeric values, Spark provides two good solutions to transform texts into indices and back.
+Since Spark can apply machine learning only on numeric values, also the labels need to be numeric.
+Fortunately, Spark offers a convenient way to map text-labels to numeric values and back.
 
-#### String Indexer 
+###### String Indexer
 
-# CONTINUE HERE
+Produces a model which automatically maps every text to a identifiable index.
+The human readable texts are stored and can be reused for convert results.
 
-#### Index To String
+###### Index To String
 
-### Machine Learning Algorithm
+Converts indices back to a human readable text. This transformer is usually applied on predictions.
+When using this transformer, it is essential to know the labels which were used.
+A String Indexer Model provides these.
 
-# Pipeline
+### Choosing a Model
 
-# Tuning
+The next step in is choosing a model.
+There are many models that researchers and data scientists have created over the years.
+Some are very well suited for image data, others for sequences (like text, or music), some for numerical data, others for text-based data.
 
-# Apps
+These two articles give a good overview to get started:
 
-# Evaluation
+* [How to choose algorithms for Microsoft Azure Machine Learning](https://docs.microsoft.com/en-us/azure/machine-learning/machine-learning-algorithm-choice)
+* [Essentials of Machine Learning Algorithms](https://www.analyticsvidhya.com/blog/2015/08/common-machine-learning-algorithms/)
 
-```
-select
-    correct.c as correct,
-    predicted.p as predicted,
-    (correct.c::double/predicted.p::double*100.0) as percent_successrate
-from
-    (select count(*) as c from predicted_tweets where label=prediction) as correct,
-    (select count(*) as p from predicted_tweets) as predicted
-```
+---
+
+For [this projects problem](https://www.burakkanber.com/blog/machine-learning-naive-bayes-1/), Naive Bayes seems to be a good and well performing solution.
```

Comment:
why

Reply:
you can also note that you did some tests with others, but would be cool if you'd have an "almost" scientific explanation why this is a good choice.